src/modules/services/n8n.nix

```
@@ -0,0 +1,86 @@
+{ config
+, lib
+, pkgs
+, ...
+}:
+
+with lib;
+
+let
+  inherit (lib) types;
+  cfg = config.services.n8n;
+  format = pkgs.formats.json { };
+  configFile = format.generate "n8n.json" cfg.settings;
+in
+{
+  options = {
+    services.n8n = {
+      enable = mkEnableOption "n8n";
+
+      address = lib.mkOption {
+        type = types.str;
+        description = "Listen address";
+        default = "127.0.0.1";
+        example = "127.0.0.1";
+      };
+
+      port = lib.mkOption {
+        type = types.port;
+        default = 5432;
+        description = ''
+          The TCP port to accept connections.
+        '';
+      };
+
+      webhookUrl = lib.mkOption {
+        type = lib.types.str;
```

If `null`, then the `WEBOOK_URL` env var won't be set at all, as long as you remove the quotes when setting the env var later.

```suggestion
        type = types.nullOr types.str;
```